libtenzir/builtins/aggregation-functions/mode_histogram.cpp

```diff
@@ -33,42 +39,62 @@ class mode_instance final : public aggregation_instance {
           continue;
         }
         ++it.value();
-        return;
       }
     }
   }
 
   auto finish() -> data override {
-    const auto comp = [](const auto& lhs, const auto& rhs) {
-      return lhs.second < rhs.second;
-    };
-    const auto it = std::ranges::max_element(counts_, comp);
-    if (it == counts_.end()) {
-      return {};
+    if constexpr (Kind == kind::mode) {
+      const auto comp = [](const auto& lhs, const auto& rhs) {
+        return lhs.second < rhs.second;
+      };
+      const auto it = std::ranges::max_element(counts_, comp);
+      if (it == counts_.end()) {
+        return {};
+      }
+      return it->first;
+    } else {
+      auto result = list{};
+      result.reserve(counts_.size());
+      for (const auto& [value, count] : counts_) {
+        result.emplace_back(record{
+          {"value", value},
+          {"count", count},
+        });
+      }
+      // TODO: Do we want to sort by frequency?
+      // std::ranges::sort(result, std::greater<>{}, [](const auto& x) {
+      //   return caf::get<int64_t>(as_vector(caf::get<record>(x))[1].second);
+      // });
```

Comment:
You'd probably want to sort by key to get a stable order, no?

Reply:
That's a good point, actually.